src/services/options.ts

```diff
@@ -18,3 +18,27 @@ export function getOption(dbPool: PostgresJsDatabase<typeof db>) {
     return res.json({ data: option });
   };
 }
+
+export function getUserOptions(dbPool: PostgresJsDatabase<typeof db>) {
+  return async function (req: Request, res: Response) {
+    const { userId } = req.params;
+
+    if (!userId) {
+      return res.status(400).json({ error: 'Missing userId' });
+    }
+
+    const query = await dbPool
+      .select()
+      .from(db.questionOptions)
+      .leftJoin(
+        db.registrationData,
+        eq(db.registrationData.id, db.questionOptions.registrationDataId),
+      )
+      .leftJoin(db.registrations, eq(db.registrations.id, db.registrationData.registrationId))
+      .where(eq(db.registrations.userId, userId));
+
+    const options = query.map((q) => q.question_options);
+
+    return res.json({ data: options });
+  };
+}
```

Comment:
There is a conflict with the following PR: https://github.com/lexicongovernance/pluraltools-backend/pull/211 that changes registrationFieldId to registrationId in the question_options table. Here, is how I think the query must be adjusted: 

```
export function getUserOptions(dbPool: PostgresJsDatabase<typeof db>) {
  return async function (req: Request, res: Response) {
    const { userId } = req.params;

    if (!userId) {
      return res.status(400).json({ error: 'Missing userId' });
    }

    const query = await dbPool
      .select()
      .from(db.questionOptions)
      .leftJoin(db.registrations, eq(db.registrations.id, db.questionOptions.registrationId))
      .leftJoin(db.registrationData, eq(db.registrationData.registrationId, db.registrations.id))
      .where(eq(db.registrations.userId, userId));

    const options = query.map((q) => q.question_options);

    return res.json({ data: options });
  };
}

```

Reply:
sounds good